### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by safar1212**
In this pull request I have implemented the following features.

Remove current_user method in ApplicationController, devise will provide us with one.
[Install and setup devise](https://github.com/heartcombo/devise#getting-started).
Can register a new user.
User logs in with a combination of email and password.
Hashed password should be stored in the database.
Ask for confirmation of email.
Can reset password.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
-  validates :Name, presence: true
-  validates :PostsCounter, numericality: { greater_than_or_equal_to: 0 }
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
 
-  def three_recent_posts
-    posts.limit(3).order(created_at: :desc)
-  end
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (ShoiraTa) [inline app/models/user.rb]: - [ ] The application should ask for confirmation of email, you can see that confirmation on rails logs and click on link. To add confirmation of email you have to add  `# :confirmable` to device

File: config/routes.rb [hunk 2]
```diff
@@ -1,9 +1,17 @@
 Rails.application.routes.draw do
-  root "users#index"
+
+  devise_for :users
+
+  devise_scope :user do
+    get '/users/sign_out' => 'devise/sessions#destroy'
+  end
```
Comment (ShoiraTa) [inline config/routes.rb]: - [ ] I can see you found a way around the delete `link_to` issue, kudos 👍🏻 
You can also solve that by making `delete` a `button_to`

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @safar1212 ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Great commit history! ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username [@shoirata].


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @safar1212 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Your linters are passing all the checks
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**